bolt-sidecar/src/state/execution.rs

```diff
@@ -311,8 +320,13 @@ impl<C: StateFetcher> ExecutionState<C> {
             return Err(ValidationError::BaseFeeTooLow(max_basefee));
         }
 
-        // Ensure max_priority_fee_per_gas is greater than or equal to min_priority_fee
-        if !req.validate_min_priority_fee(max_basefee, self.limits.min_priority_fee) {
+        // Ensure max_priority_fee_per_gas is greater than or equal to the calculated min_priority_fee
+        if !req.validate_min_priority_fee(
+            &self.pricing,
+            template_committed_gas,
+            self.limits.min_inclusion_profit,
+            max_basefee,
+        )? {
             return Err(ValidationError::MaxPriorityFeePerGasTooLow);
```

Comment:
We should return the value of the minimum fee in this error, so it can be propagated to any consumers